flag_engine/identities/traits/types.py

```diff
@@ -1,13 +1,57 @@
-from typing import Union
+import re
+from decimal import Decimal
 
-from pydantic.types import confloat, constr
+from typing import Union, Any, get_args
+from typing_extensions import TypeGuard
+
+from pydantic.types import (
+    AllowInfNan,
+    StringConstraints,
+    StrictBool,
+)
+from pydantic import BeforeValidator
+from typing_extensions import Annotated
 
 from flag_engine.identities.traits.constants import TRAIT_STRING_VALUE_MAX_LENGTH
 
-TraitValue = Union[
-    None,
-    int,
-    confloat(allow_inf_nan=False),
-    bool,
-    constr(max_length=TRAIT_STRING_VALUE_MAX_LENGTH),
+_UnconstrainedTraitValue = Union[None, int, float, bool, str]
+
+
+def map_any_value_to_trait_value(value: Any) -> _UnconstrainedTraitValue:
```

Comment:
I think some documentation on this function would be good.

Reply:
Done. Also made this function private.